support-frontend/assets/pages/contributions-landing/contributionsLandingActions.js

```diff
@@ -350,40 +349,54 @@ const stripeChargeDataFromPaymentIntentAuthorisation = (
   state,
 );
 
-const regularPaymentRequestFromAuthorisation = (
+function regularPaymentRequestFromAuthorisation(
   authorisation: PaymentAuthorisation,
   state: State,
-): RegularPaymentRequest => ({
-  firstName: state.page.form.formData.firstName || '',
-  lastName: state.page.form.formData.lastName || '',
-  email: state.page.form.formData.email || '',
-  billingAddress: {
-    lineOne: null, // required go cardless field
-    lineTwo: null, // required go cardless field
-    city: null, // required go cardless field
-    state: state.page.form.formData.state,
-    postCode: null, // required go cardless field
-    country:
-      state.page.form.formData.billingCountry ?
-        state.page.form.formData.billingCountry : state.common.internationalisation.countryId,
-  },
-  deliveryAddress: null,
-  product: {
-    amount: getAmount(
-      state.page.form.selectedAmounts,
-      state.page.form.formData.otherAmounts,
-      state.page.form.contributionType,
-    ),
-    currency: state.common.internationalisation.currencyId,
-    billingPeriod: state.page.form.contributionType === 'MONTHLY' ? Monthly : Annual,
-  },
-  firstDeliveryDate: null,
-  paymentFields: regularPaymentFieldsFromAuthorisation(authorisation),
-  ophanIds: getOphanIds(),
-  referrerAcquisitionData: state.common.referrerAcquisitionData,
-  supportAbTests: getSupportAbTests(state.common.abParticipations),
-  telephoneNumber: null,
-});
+): RegularPaymentRequest {
+
+  let { billingCountry, billingState } = state.page.form.formData;
```

Comment:
Not sure it's any better but I had a go at a version without vars:
```
function getBillingCountryAndState(state: State): {
  billingCountry: IsoCountry,
  billingState: string | null,
} {
  if (state.page.form.formData.billingCountry) {
    return { billingCountry, billingState } = state.page.form.formData;
  } else {
    return {
      billingCountry:
        findIsoCountry(window.guardian.geoip.countryCode) ||
        // if GEO-IP fails to resolve a country, then get it from the page's internationalisation which will never be null.
        state.common.internationalisation.countryId,
      // if we're going to be taking the country from GEO-IP then we need to take the state from there too
      billingState: stateProvinceFromString(billingCountry, window.guardian.geoip.stateCode)
    }
  }
}
```
and then:
`const { billingCountry, billingState } = getBillingCountryAndState(state);`

Reply:
I've done something very similar. Flow didn't like not using const in destructuring, so I just pulled the relevant values out from `formData` the old fashioned way.